java/gateway-core/src/test/java/com/avaulta/gateway/rules/MultiTypeBulkDataRulesTest.java

```diff
@@ -0,0 +1,82 @@
+package com.avaulta.gateway.rules;
+
+import com.avaulta.gateway.rules.transforms.RecordTransform;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.Test;
+
+import java.io.InputStream;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class MultiTypeBulkDataRulesTest {
+
+
+    private final ObjectMapper yamlMapper = new ObjectMapper(new YAMLFactory());
+
+    @SneakyThrows
+    @Test
+    public void fromYaml() {
+
+        InputStream yamlStream = getClass().getClassLoader().getResourceAsStream("rules/multi-type-example.yaml");
+
+
+        MultiTypeBulkDataRules rules = yamlMapper.readerFor(MultiTypeBulkDataRules.class).readValue(yamlStream);
+
+        // Basic checks to ensure deserialization worked (you can add more specific checks)
+        assertNotNull(rules);
+        RecordRules record = (RecordRules) rules.getFileRules().get("/export/{week}/index_{shard}.ndjson");
+        assertEquals(RecordRules.Format.valueOf("NDJSON"), record.getFormat());
+        assertNotNull(record);
+        assertFalse(record.getTransforms().isEmpty());
+    }
+
+    @SneakyThrows
+    @Test
+    public void toYaml() {
+
+        final String EXPECTED = "---\n" +
+            "fileRules:\n" +
+            "  /export/{week}/data_{shard}.csv:\n" +
+            "    columnsToPseudonymize:\n" +
+            "    - \"email\"\n" +
+            "    delimiter: \",\"\n" +
+            "    pseudonymFormat: \"JSON\"\n" +
+            "  /export/{week}/index_{shard}.ndjson:\n" +
```

Comment:
:warning: so we are not expecting gzipped content??? should we account for that too? like `export/{week}/index_{shard}.ndjson.gz` ?
I think we should, and explicitly put format = CSV or format = NDJSON always to choose one or another
and reader check if gzipped or not.

Reply:
these are just example rules. I have a task to test out `gzip` on separate branch. I think for GCS it will just magically work; when you open the file stream, GCS client will by default detect compression and decompress it (unless you disable this behavior) - but need to test that. 

And then need to write compressed content back, which we currently don't support - but is certainly worth doing.